lib/puppet/provider/cert_file/posix.rb

```diff
@@ -0,0 +1,63 @@
+require 'pp'
+
+Puppet::Type.type(:cert_file).provide :posix do
+  confine feature: :posix
+
+  def exists?
+    return false unless Pathname.new(resource[:path]).exist?
+
+    debug "Checking file format #{resource[:path]}"
+    localcert_file = File.read(resource[:path])
+    unless identify_cert_format(localcert_file) == resource[:format]
```

Comment:
Since `identify_cert_format` is only used for this code, maybe turn it into a `is_cert_format?` method and pass both the file and the format and return a boolean?

Maybe also make it a simpler short-circuit, like:

```ruby
return false unless is_cert_format?(localcert_file, resource[:format])
```

Reply:
Sounds reasonable.